tests/data_extraction/test_mapping_script.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+import rdflib
+import datetime
+import os
+import yaml
+
+from pathlib import Path
+
+from lebedigital.mapping.emodul_mapping import generate_knowledge_graph, get_date_time_value
+
+def query_and_test_result(typeOfQuery, predicate, prop, g):
+    print(f"""Query for {typeOfQuery}:""")
+    q = f"""
+            SELECT ?o WHERE {{
+                    {predicate} {prop} ?o .
+            }}
+            """
+    print(q)
+    result = g.query(q)
+    return result
+      
+def print_next():
+    print("Pass")
+    print("###################Next Query#####################################")
+
+def print_first():
+    print("###################First Query####################################")
+
+def test_generate_knowledge_graph():
+    metadataPath = 'testMetaData.yaml'
+    filename = "knowledgeGraph.ttl"
+    print("Start testing:")
+    #Generate a testMetadata file
+    print("Generate metadata file for test:")
+    target_data = {'experimentName': 'BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4',
+            'software_specification': 'MTS793|MPT|DEU|1|2|,|.|:|49|1|1|A',
+            'operator_timestamp': '13:25:39',
+            'operator_date': '01.09.2014',
+            'tester_name': 'Kh',
+            'specimen_name': 'BA-Losert E-Modul 28d v. 04.08.14 Probe 4',
+            'remark': 'Kraftgeregelt 3,9 kN/s',
+            'weight': 5342.0,
+            'weight_unit': 'g',
+            'diameter': 98.6,
+            'length': 300.3,
+            'length_unit': 'mm',
+            'mix_file': '2014_08_05 Rezeptur_MI.xlsx'}
+
+    with open(metadataPath, 'w') as yamlFile:
+        yaml.dump(target_data, yamlFile)
+    print("Generate knowledgeGraph:")
+    generate_knowledge_graph(metadataPath, filename)
+    
+    g = rdflib.Graph()
+    result = g.parse(filename, format='ttl')
+    print("Query knowledgeGraph:")
+    
+    print_first()
+
+    result = query_and_test_result('diameter','ns1:informationbearingentity1', 
```

Comment:
Wouldn't it be much more important to query the diameter for a given experiment (thus traversing the knowledge graph) instead of checking a single triple?

Reply:
I don't really know how to traverse the knowledge graph using SPARQL queries but I will get in touch with Andre and try to apply his suggestion in my tests. I may be interesting to discuss about a standard nomenclature that I could use for getting rid of names like informationbearingentity1 that are generated automatically by owlready. Maybe something like nameOfExperimentDiameter or nameOfSpecimentDiameter.

Reply:
Getting rid of informationbearingentity1  is tricky since this is part of the class structure prescribed by the ontology (not by owlready), i.e. this is something you should discuss with @StephanPirskawetz  and @firmao together. As for the complete sparql query, please talk to @firmao .